evalml/utils/woodwork_utils.py

```diff
@@ -27,6 +28,32 @@ def _list_to_pandas(list):
     return _numpy_to_pandas(np.array(list))
 
 
+def _new_infer_boolean_nullable_func(series):
```

Comment:
Even though this function is private, context on what it's doing would be useful for future work. Could you add a docstring?